### PR TITLE
Fix godep verify to use godep restore script

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
A bug was introduced in #48615. `hack/verify-godeps.sh` only downloads and compares if godeps have changed, so it wasn't caught on the original PR. However, when it does run (e.g. https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/48630/pull-kubernetes-verify/38350/) it fails because the godep-save script now doesn't permit a compex GOPATH. verify-godeps.sh actually sets one because it restores godeps not using the `hack/godep-restore.sh` script.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/assign @sttts
/priority failing-test

### CODE REVIEW

**Comment by sttts** [comment]
@eparis approved?

**Comment by cblecker** [comment]
.

File: hack/verify-godeps.sh [hunk 1]
```diff
@@ -48,7 +48,8 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 readonly branch=${1:-${KUBE_VERIFY_GIT_BRANCH:-master}}
 if ! [[ ${KUBE_FORCE_VERIFY_CHECKS:-} =~ ^[yY]$ ]] && \
   ! kube::util::has_changes_against_upstream_branch "${branch}" 'Godeps/' && \
-  ! kube::util::has_changes_against_upstream_branch "${branch}" 'vendor/'; then
+  ! kube::util::has_changes_against_upstream_branch "${branch}" 'vendor/' && \
+  ! kube::util::has_changes_against_upstream_branch "${branch}" 'hack/'; then
```
Comment (liggitt) [inline hack/verify-godeps.sh]: this adds ~15 minutes to the verify job for any PR that touches *anything* in `hack`... can we target this to things that touched a godep-related script under hack?

Comment (liggitt) [inline hack/verify-godeps.sh]: opened https://github.com/kubernetes/kubernetes/pull/50418

Comment (sttts) [inline hack/verify-godeps.sh]: What about caching? @cblecker didn't we introduce a way for caching?

**Pull request merged**